api/src/main/java/org/apache/iceberg/metrics/MetricsContext.java

```diff
@@ -51,64 +50,6 @@ public static Unit fromDisplayName(String displayName) {
 
   default void initialize(Map<String, String> properties) {}
 
-  /** @deprecated Use {@link org.apache.iceberg.metrics.Counter} instead. */
-  @Deprecated
```

Comment:
I'd love to get rid of these, but the deprecation happened [after 0.14.0](https://github.com/apache/iceberg/commit/747d2b0ef1e422f9aac1bd61a3fc46bb84c305b6#), so we shouldn't remove it until 2.0.0 because it hasn't been deprecated in a release yet.